Include/internal/pycore_interp.h

```diff
@@ -168,8 +171,25 @@ struct _is {
 
     struct ast_state ast;
     struct type_cache type_cache;
+
+    struct {
+        // See _PyRuntimeState._preallocated for an explanation.
+        PyThreadState tstate;
+        // XXX Pre-allocate as many objects from above as possible here.
+    } _preallocated;
```

Comment:
Please remove the `_preallocated` struct.
This thread state is always the main thread, right?
So you could replace this struct with `PyThreadState main_thread;` which better expresses what it is.

Reply:
I'd also rather keep the struct, to distinguish these fields from the corresponding pointer fields.  (See https://github.com/python/cpython/pull/29883#discussion_r781603894.)

Also, we will be adding more fields to `_preallocated`, so it won't just stay a struct with a single field.

Reply:
As to the `tstate` field name, I considered a number of names for it, including "main_thread".  However, "main_thread" is confusing because there is only one main thread in the process, where signals are handled, and it corresponds only to the initial thread of the main interpreter.

Regardless, I'd be fine with a more descriptive name.  The name "initial_thread" might be better.

Reply:
"initial_thread" sounds good.